dask_ndfourier/_compat.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+
+
+import itertools
+
+import numpy
+
+import dask.array
+
+
+def _indices(dimensions, dtype=int, chunks=None):
+    """
+    Implements NumPy's ``indices`` for Dask Arrays.
+    Generates a grid of indices covering the dimensions provided.
+    The final array has the shape ``(len(dimensions), *dimensions)``. The
+    chunks are used to specify the chunking for axis 1 up to
+    ``len(dimensions)``. The 0th axis always has chunks of length 1.
+
+    Parameters
+    ----------
+    dimensions : sequence of ints
+        The shape of the index grid.
+    dtype : dtype, optional
+        Type to use for the array. Default is ``int``.
+    chunks : sequence of ints
+        The number of samples on each block. Note that the last block will
+        have fewer samples if ``len(array) % chunks != 0``.
+
+    Returns
+    -------
+    grid : dask array
+
+    Notes
+    -----
+    Borrowed from my Dask Array contribution.
```

Comment:
Please see PR ( https://github.com/dask/dask/pull/2268 ) for details. Also borrowed our tests from there as well.